concordium-contracts-common/src/types.rs

```diff
@@ -677,6 +723,25 @@ impl ContractAddress {
     }
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for ContractAddress {
+    fn arbitrary(g: &mut Gen) -> ContractAddress {
+        ContractAddress {
+            index:    quickcheck::Arbitrary::arbitrary(g),
+            subindex: quickcheck::Arbitrary::arbitrary(g),
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        let index_shrink = quickcheck::Arbitrary::shrink(&self.index);
+        let subindex_shrink = quickcheck::Arbitrary::shrink(&self.subindex);
+        Box::new(index_shrink.zip(subindex_shrink).map(|(i, si)| ContractAddress {
+            index:    i,
+            subindex: si,
+        }))
```

Comment:
What is the idea here?

Don't we want something like a product of those two iterators? Why is zipping them the sensible thing?

Reply:
This comes from Simon's code. I agree, it should be a product. Is there an std function that creates a product of iterators? I could only find an external library https://docs.rs/itertools/0.10.5/itertools/structs/struct.Product.html

Reply:
Yes, fair enough, I'll use nested loops. Just thought that maybe something standard is available, so I don't reinvent the wheel :)

Reply:
Done here an for `OwnedPolicy` as well.